packages/runtime/test/src/interpreter/opcode-list.ts

```diff
@@ -3607,4 +3608,147 @@ describe("Teal Opcodes", function () {
+
+      stack.push(new Uint8Array([1, 2, 3, 4, 5])); // target
+      stack.push(2n); // index
+
+      op.execute(stack);
+      assert.equal(stack.pop(), 3n);
+    });
+
+    it("should panic if target is not bytes", () => {
+      const op = new GetByte([], 0);
+      stack.push(10n); // target
+      stack.push(0n); // index
+
+      expectRuntimeError(
+        () => op.execute(stack),
+        RUNTIME_ERRORS.TEAL.INVALID_TYPE
+      );
+    });
+
+    it("should panic if index is not uint", () => {
+      const op = new GetByte([], 0);
+      stack.push(new Uint8Array(0)); // target
+      stack.push(new Uint8Array(0)); // index
+
+      expectRuntimeError(
+        () => op.execute(stack),
+        RUNTIME_ERRORS.TEAL.INVALID_TYPE
+      );
+    });
+
+    it("should panic if index bit in out of bytes array", () => {
+      const op = new GetByte([], 0);
+      stack.push(new Uint8Array(0)); // target
+      stack.push(500n); // index
+
+      expectRuntimeError(
+        () => op.execute(stack),
+        RUNTIME_ERRORS.TEAL.SET_BIT_INDEX_BYTES_ERROR
+      );
+
+      stack.push(new Uint8Array(5).fill(0)); // target
+      stack.push(64n * 5n + 1n); // index
+
+      expectRuntimeError(
+        () => op.execute(stack),
+        RUNTIME_ERRORS.TEAL.SET_BIT_INDEX_BYTES_ERROR
+      );
+    });
+  });
+
+  describe("SetByte", () => {
+    let stack: Stack<StackElem>;
+    this.beforeEach(() => { stack = new Stack<StackElem>(); });
+
+    it("should set correct bytes in stack", () => {
+      const op = new SetByte([], 0);
+      stack.push(new Uint8Array([8, 2, 1, 9])); // target
+      stack.push(0n); // index
+      stack.push(5n); // small integer
+
+      op.execute(stack);
+      assert.deepEqual(stack.pop(), new Uint8Array([5, 2, 1, 9]));
+
+      stack.push(new Uint8Array([8, 2, 1, 9])); // target
+      stack.push(3n); // index
+      stack.push(0n); // small integer
+
+      op.execute(stack);
+      assert.deepEqual(stack.pop(), new Uint8Array([8, 2, 1, 0]));
+    });
+
+    it("should panic if target is not bytes", () => {
+      const op = new SetByte([], 0);
+      stack.push(1n); // target
+      stack.push(0n); // index
+      stack.push(12n);
+
+      expectRuntimeError(
+        () => op.execute(stack),
+        RUNTIME_ERRORS.TEAL.INVALID_TYPE
+      );
+    });
+
+    it("should panic if index of small integer is not uint", () => {
+      const op = new SetByte([], 0);
+      stack.push(new Uint8Array(0)); // target
+      stack.push(new Uint8Array(0)); // index
+      stack.push(12n);
+
+      expectRuntimeError(
+        () => op.execute(stack),
+        RUNTIME_ERRORS.TEAL.INVALID_TYPE
+      );
+
+      stack.push(new Uint8Array(0)); // target
+      stack.push(1n); // index
+      stack.push(new Uint8Array(0));
+
+      expectRuntimeError(
+        () => op.execute(stack),
+        RUNTIME_ERRORS.TEAL.INVALID_TYPE
+      );
+    });
+
+    it("should panic if index bit in out of bytes array", () => {
```

Comment:
could you please check if this line is correct?

Reply:
ok